### ISSUE TITLE
docker build with DOCKER_BUILDKIT=1 unable to pull armhf images from multi-arch manifest

### ISSUE THREAD

**Issue by andrewhsu** [moby/moby#37631]
**Description**

docker build with `DOCKER_BUILDKIT=1` unable to pull armhf image from multi-arch manifest

**Steps to reproduce the issue:**
With `Dockerfile`:
```
FROM alpine:latest
```

Do `docker build` with [experimental](https://docs.docker.com/engine/reference/commandline/dockerd/#description) enabled and `DOCKER_BUILDKIT=1`.

**Describe the results you received:**

With `DOCKER_BUILDKIT=1`, does not complete successfully:
```
$ DOCKER_BUILDKIT=1 docker build .
[+] Building 2.6s (4/4) FINISHED                                                                                               
 => [internal] load .dockerignore                                                                                         0.4s
 => => transferring context: 02B                                                                                          0.0s
 => [internal] load Dockerfile                                                                                            0.5s
 => => transferring dockerfile: 130B                                                                                      0.0s
 => ERROR [internal] load metadata for docker.io/library/alpine:latest                                                    1.4s
 => CANCELED [1/1] FROM docker.io/library/alpine:latest                                                                   0.5s
 => => resolve docker.io/library/alpine:latest                                                                            0.5s
------
 > [internal] load metadata for docker.io/library/alpine:latest:
------
no match for current platform linux/arm/v7 in manifest sha256:7043076348bf5040220df6ad703798fd8593a0918d06d3ce30c6c93be117e430: not found
```

**Describe the results you expected:**

It works with legacy `build`:
```
$ docker build .
Sending build context to Docker daemon  6.804MB
Step 1/1 : FROM alpine:latest
latest: Pulling from library/alpine
ee7d700abbf2: Pull complete 
a1653f4692c1: Pull complete 
Digest: sha256:7043076348bf5040220df6ad703798fd8593a0918d06d3ce30c6c93be117e430
Status: Downloaded newer image for alpine:latest
 ---> 4407218c8304
Successfully built 4407218c8304
```

**Additional information you deem important (e.g. issue happens only occasionally):**

For reference, output of `docker manifest inspect alpine:latest`:
```json
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 528,
         "digest": "sha256:0873c923e00e0fd2ba78041bfb64a105e1ecb7678916d1f7776311e45bf5634b",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 735,
         "digest": "sha256:78f3ccd48cc6a55709b65c8fdb3ef81ed922c5393b064d63b0d35f51e0c9fb2e",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v6"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 735,
         "digest": "sha256:30ebbffae9ba098b9c345b890c6610aa8422f067754d7e707978a954f3b7a601",
         "platform": {
            "architecture": "arm64",
            "os": "linux",
            "variant": "v8"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 735,
         "digest": "sha256:5d757fdef3d336720bbccfbaa6e32089acab8ade35992e14bb0253111beb3026",
         "platform": {
            "architecture": "386",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 735,
         "digest": "sha256:a332d6595c30bee40576e8cb1c75fc9ea7e646b6d159bb16cd1858085280aad9",
         "platform": {
            "architecture": "ppc64le",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 735,
         "digest": "sha256:903e1eeceeb7208949673dcb557659b5f4c8e9065a45432a0668acdd0069268c",
         "platform": {
            "architecture": "s390x",
            "os": "linux"
         }
      }
   ]
}
```

**Output of `docker version`:**

```
Client:
 Version:           0.0.0-20180813050154-b04c87d
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        b04c87d
 Built:             Mon Aug 13 07:14:55 2018
 OS/Arch:           linux/arm
 Experimental:      false

Server:
 Engine:
  Version:          0.0.0-20180813050154-b04c87d
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       b04c87d
  Built:            Mon Aug 13 07:10:37 2018
  OS/Arch:          linux/arm
  Experimental:     true
```

**Output of `docker info`:**

```
...
Experimental: true
...
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

```
$ uname -mpi
armv7l armv7l armv7l
```

cc @tonistiigi @tiborvass

**Comment by tonistiigi** [comment]
This is a containerd platform matching problem that we have discussed in the past. There is no `v7` alpine in manifest list, the current platform is `v7` and containerd enforces variant to match. There are some more compatibility rules as well for other chips. @dmcgowan

**Comment by tonistiigi** [comment]
@andrewhsu This should be fixed in `18.09`. Can you verify?

**Comment by andrewhsu** [comment]
Yup, stuff works with `18.09.0`:
```
# DOCKER_BUILDKIT=1 docker build .
[+] Building 0.8s (5/5) FINISHED                                                                                        
 => [internal] load build definition from Dockerfile                                                               0.2s
 => => transferring dockerfile: 56B                                                                                0.0s
 => [internal] load .dockerignore                                                                                  0.4s
 => => transferring context: 2B                                                                                    0.0s
 => [internal] load metadata for docker.io/library/alpine:latest                                                   0.0s
 => [1/1] FROM docker.io/library/alpine:latest                                                                     0.2s
 => => resolve docker.io/library/alpine:latest                                                                     0.0s
 => exporting to image                                                                                             0.0s
 => => exporting layers                                                                                            0.0s
 => => writing image sha256:c265a2bdb3571f053b1dc7c7c696a00f0f4a54fd77b07c6a65a7ea82499351d4                       0.0s
# docker version
Client:
 Version:           18.09.0
 API version:       1.39
 Go version:        go1.10.4
 Git commit:        4d60db4
 Built:             Wed Nov  7 00:56:41 2018
 OS/Arch:           linux/arm
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.0
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.4
  Git commit:       4d60db4
  Built:            Wed Nov  7 00:17:57 2018
  OS/Arch:          linux/arm
  Experimental:     true
```
